modules/SampleModule/test/test_samplemodule.cpp

```diff
@@ -0,0 +1,113 @@
+#include "gtest/gtest.h"
+#include "rclcpp/rclcpp.hpp"
+#include "tests/rclcpp_utils.hpp"
+#include "samplemodule.hpp"
+class SampleModuleTest : public ::testing::Test {
+public:
+  static void SetUpTestCase()
+  {
+    rclcpp::init(0, nullptr);
+    
+  }
+
+  static void TearDownTestCase()
+  {
+
+
+    rclcpp::shutdown();
+  }
+
+  protected:
+
+  std::shared_ptr<SampleModule> sampleModule;
+  std::shared_ptr<rclcpp::Node> helperNode;
+  rclcpp::executors::SingleThreadedExecutor executor;
+
```

Comment:
What is the differences between SetUp() and SetUpTestCase()?

Reply:
Setup/Teardown runs before/after every test. SetupTestCase/TearDownTestCase runs before and after the entire set of tests